Gems/PhysX/Code/Source/EditorShapeColliderComponent.cpp

```diff
@@ -849,4 +939,46 @@ namespace PhysX
     {
         return m_colliderConfig.m_isTrigger;
     }
+
+    void EditorShapeColliderComponent::UpdateTriggerSettings()
+    {
+        if (m_shapeType == ShapeType::QuadSingleSided || m_shapeType == ShapeType::QuadDoubleSided)
+        {
+            if (!m_previousIsTrigger.has_value())
+            {
+                m_previousIsTrigger = m_colliderConfig.m_isTrigger;
+                m_colliderConfig.m_isTrigger = false;
+            }
+            m_colliderConfig.SetPropertyVisibility(Physics::ColliderConfiguration::PropertyVisibility::IsTrigger, false);
+        }
+        else
+        {
+            if (m_previousIsTrigger.has_value())
+            {
+                m_colliderConfig.m_isTrigger = m_previousIsTrigger.value();
+                m_previousIsTrigger.reset();
+            }
+            m_colliderConfig.SetPropertyVisibility(Physics::ColliderConfiguration::PropertyVisibility::IsTrigger, true);
+        }
+    }
+
+    void EditorShapeColliderComponent::UpdateSingleSidedSettings()
+    {
+        if (GetEntity()->FindComponent<EditorRigidBodyComponent>())
```

Comment:
aside: I wish there was a `HasComponent` API instead... (not really related to the PR just an observation)

Reply:
Yeah, that would be nicer.